rewriter/TEnum.cc

```diff
@@ -105,14 +105,14 @@ vector<unique_ptr<ast::Expression>> processStat(core::MutableContext ctx, ast::C
             return badConst(ctx, stat->loc, klass->loc);
         }
 
-        if (!(arg0->fun == core::Names::new_() && arg0->recv->isSelfReference())) {
+        if (!ast::MK::isSelfNew(arg0)) {
```

Comment:
Why did the `isSelfReference` drop here? In the condition above it was replaced with `isMagicClass`

Reply:
The conditionals are different: in the previous case it's checking if the function is `new`, and the receiver is not `Magic`, while the second is checking the case that the functions is `new` and the receiver is `Magic`.

Reply:
ty again!